### DISCUSSION TITLE
https_redirect not listening on port 80?

### DISCUSSION THREAD

**Discussion by mombooks** [drakkan/sftpgo#1170] [Q&A] [answered]
I've got port set to 443, security enabled set to true and https_redirect set to true, but I get can't connect to server when typing http:// instead of https://

I've reloaded and restarted the service and rebooted the server but no luck. Am I missing some config? I feel like maybe it's because only port 443 is specified

**Comment by drakkan** [comment, marked as answer]
Hello,

you have to define an additional HTTP binding on port 80 and enable the redirect for this binding.
The problem is that this way certificate renewal will not work because when SFTPGo try to renew the certificate it will fail because port 80 is already in use.

**Comment by mombooks** [comment]
Ah thanks, I'l leave it as is then as auto-renewal is more important for us. If there's any way to make both work together in a future release that would be amazing!

**Reply by drakkan** [reply]
> Ah thanks, I'l leave it as is then as auto-renewal is more important for us. If there's any way to make both work together in a future release that would be amazing!

I'll consider whether it's worth adding anything to make this use case possible. 

With a standard web server (NGINX, Apache) listening on port 80 you can redirect SFTPGo requests to https and at the same time you can serve the acme challenge reponses using the web root method or a reverse proxy.

**Reply by drakkan** [reply]
> Ah thanks, I'l leave it as is then as auto-renewal is more important for us. If there's any way to make both work together in a future release that would be amazing!

https://github.com/drakkan/sftpgo/commit/9a1074021843b2a3e94d33c00f437bbafd035094

this change will be included in v2.5.0

**Comment by mombooks** [comment]
That's a good idea but I think I'd have to spend some time tinkering to get it working and possibly break things in the process, unfortunately we've had to go live with this server rather quickly as our old FTP server had a limit of 8192 links it could store in the database and we'd run out!
